src/jni/org/netbsd/liblpm/LPM.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2016 Henry Rodrick <henry at holodisc org uk>
+ * All rights reserved.
+ *
+ * Use is subject to license terms, as specified in the LICENSE file.
+ */
+
+package org.netbsd.liblpm;
+
+import java.io.*;
+
+public class LPM<T> {
+	static {
+		File temp = null;
+		try {
+			final String name = "org_netbsd_liblpm_LPM.so";
+			InputStream in = LPM.class.getResourceAsStream("/" + name);
+			byte[] buffer = new byte[1024];
+			int read = -1;
+			temp = File.createTempFile(name, "");
+			FileOutputStream fos = new FileOutputStream(temp);
+
+			while((read = in.read(buffer)) != -1) {
+				fos.write(buffer, 0, read);
+			}
+			fos.close();
+			in.close();
+
+			System.load(temp.getAbsolutePath());
+		} catch (IOException e) {
+			throw new RuntimeException(e);
+		}
+		if (temp != null) {
+			temp.delete();
```

Comment:
This should happen in a finally-block. Actually, the whole think is a bit of a copy-paste and I should clean up the stream handling.